tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/ESEmailAddressContactSearchEngine.java

```diff
@@ -0,0 +1,19 @@
+package com.linagora.tmail.james.jmap.model;
+
+import org.apache.james.jmap.api.model.AccountId;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+import scala.runtime.BoxedUnit;
+
+public class ESEmailAddressContactSearchEngine implements EmailAddressContactSearchEngine{
+    @Override
+    public Publisher<BoxedUnit> index(AccountId accountId, EmailAddressContact contact) {
+        //Mono.fromCallable()
+        return null;
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String part) {
+        return null;
```

Comment:
Here you might need to use maybe the analyzer? https://www.elastic.co/guide/en/elasticsearch/client/java-rest/7.10/java-rest-high-analyze.html

I would see a side PR on apache James adding the analyze method in ReactorElasticSearchClient so that you can reuse it here. 

@chibenwa WDYT?

Reply:
We tend to rely on the indexer definition on the indexed fields (data is stored structured to answer the analyzer), I do not feel confident having this specified in a per request basis as it is more complicated to assess that the data stored is structured the right way.

How about defining a mapping with the right analyzer instead?